src/load.rs

```diff
@@ -67,11 +67,11 @@ impl Loader {
 
     /// Convert a path string to a FileId.  For performance reasons
     /// this requires an owned 'path' param.
-    fn path(&mut self, path: String) -> FileId {
+    fn path(&mut self, mut path: String) -> FileId {
         // Perf: this is called while parsing build.ninja files.  We go to
         // some effort to avoid allocating in the common case of a path that
         // refers to a file that is already known.
-        let path = canon_path_fast(path);
+        canon_path_fast(&mut path);
```

Comment:
It looks like this could just call canon_path now, I think?  It looks like that will take the input string and mutate it, won't it?